app/models/miq_server/server_monitor.rb

```diff
@@ -19,17 +19,21 @@ def make_master_server(last_master)
     _log.info "Master server has #{last_master.nil? ? "not been set" : "died, #{last_master.name}"}.  Attempting takeover as new master server, #{name}."
     parent = MiqRegion.my_region
     parent.lock do
-      all_servers = parent.miq_servers
+      # See if an ACTIVE server has already taken over
+      active_servers = parent.active_miq_servers
 
       _log.debug "Double checking that nothing has changed"
-      master = all_servers.detect(&:is_master?)
+      master = active_servers.detect(&:is_master?)
       if (last_master.nil? && !master.nil?) || (!last_master.nil? && !master.nil? && last_master.id != master.id)
         _log.info "Aborting master server takeover as another server, #{master.name}, has taken control first."
         return nil
       end
 
-      all_servers.each do |s|
       _log.debug "Setting this server, #{name}, as master server"
+
+      # Set is_master on self, reset every other server in the region, including
+      # inactive ones.
+      parent.miq_servers.each do |s|
```

Comment:
So this is to ensure that we properly set the is_master on all servers. Even the ones that are inactive, right?

Also, is there a chance that `miq_servers` could have been cached? Should we do `MiqRegion.my_region(true)` above when `parent` is set? Or `parent.reload.miq_servers` here.